### ISSUE TITLE
[rush] Allow "publish.npmrc" instead of ".npmrc-publish"

### ISSUE THREAD

**Issue by AndrewCraswell** [microsoft/rushstack#2248]
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [x] Feature
- [ ] Bug

**Please describe the actual behavior.**

Publish settings currently must be set in the `.npmrc-publish` file. However, when our Azure Pipeline is executing, we use the `npmAuthenticate` task to authenticate the .npmrc with a temp token from the build agent, which prevents us from having to use a manually generated token which expires.

The problem is that the `npmAuthenticate` task has an explicit check to verify that the extension of the file is exactly ".npmrc". Since Rush uses a nonstandard name for the publish file, it's not certain to work in some pipelines when using the documented best practices.

**What is the expected behavior?**

Either the `.npmrc-publish` file should be renamed to 'publish.npmrc' or either should be accepted (for backward compatibility).

**Comment by iclanton** [comment]
Seems like a good idea. Would you be interested in raising a PR?

>Either the `.npmrc-publish` file should be renamed to 'publish.npmrc' or either should be accepted (for backward compatibility).

I'd say both should be accepted for now, and then we can drop `.npmrc-publish` in the next major release. @octogonz - sound good to you?

**Comment by octogonz** [comment]
> I'd say both should be accepted for now, and then we can drop `.npmrc-publish` in the next major release. @octogonz - sound good to you?

I'm okay with that.

But with the caveat that, in my view, the `npmAuthenticate` task is a security concern.  It writes an auth token into a text file.  Thus if an unrelated CI job is running concurrently on the same machine, it can read the token.  Even if it's running at a later time, it can read the token if the file was not deleted for some reason (e.g. the machine rebooted unexpectedly).

Our recommendation is to pass the token via an environment variable rather than a text file.  It seems the ideal solution would be for  Azure to improve the `npmAuthenticate` task to support environment variables.

**Comment by AndrewCraswell** [comment]
I agree with the idea that `npmAuthenticate` could be improved. I've logged another issue for that bit.
https://github.com/microsoft/azure-pipelines-tasks/issues/13646
